fe/user/src/pages/MapPage/Panel/VenueDetail/RestInfo/ShowInfo/ShowList.tsx

```diff
@@ -19,6 +21,7 @@ export const ShowList: React.FC<Props> = ({
   selectPreviousDate,
   selectNextDate,
 }) => {
+  const { showId, setShowId } = useShowDetailStore();
```

Comment:
zustand 셀렉터 기능을 사용하면 렌더링 최적화를 꾀할 수 있어요.

```suggestion
  const showId = useShowDetailStore((state) => state.showId);
  const setShowId = useShowDetailStore((state) => state.setShowId);
```

`store`를 여러번 호출하는 것이 싫다면 한줄로 값을 받아올 수 있습니다. 한번에 호출로 가져오려면 `useShallow`라는 메서드를 같이 사용해주어야 합니다. 왜냐하면 zustand가 이전 데이터를 비교할 때 동등 연산자(`===`)와 같이 엄격하게 비교하기 때문에 얕은 비교를 하게끔 해야하기 때문입니다.
[참조 링크](https://github.com/pmndrs/zustand?tab=readme-ov-file#selecting-multiple-state-slices)

```suggestion
  const { showId, setShowId } = useShowDetailStore(useShallow((state) => ({ showId: state.showId, setShowId: state.setShowId })));
```

Reply:
`useShallow`로 얕은 비교를 하니 관심있는 `state`가 변경될 때만 재렌더링이 일어나는 거군요!
적용을 해봤더니 ShowInfo 컴포넌트의 재렌더링이 10회에서 6회로 줄어들었어요. 👍
불필요한 재렌더링이 4회나 발생하고 있었다니...쿤디 덕분에 수정할 수 있었어요.
https://github.com/jazz-meet/jazz-meet/pull/289/commits/59d057130d1a95ad05b0e3e418a58bc67f2318de